### PR TITLE
Check all invariants across multiple distribution periods

### PR DESCRIPTION
**Pull request by MikeHathaway**
# Description of change
## High level
* Update invariants to work in the `invariant_all` scenario across multiple distribution periods.
  * Add support for checking screening stage and funding stage invariants across multiple distribution periods

### CODE REVIEW

File: test/invariants/INVARIANTS.md [hunk 1]
```diff
@@ -22,12 +22,13 @@
     - **SS4**: Screening vote's cast can only be positive.
     - **SS5**: Screening votes can only be cast on a proposal in it's distribution period's screening stage.
     - **SS6**: For every proposal, it is included in the top 10 list if, and only if, it has as many or more votes as the last member of the top ten list (typically the 10th of course, but it may be shorter than ten proposals).
+    - **SS7**: Screening vote's on a proposal should cause addition to the topTenProposals if the array is unpopulated.
```
Comment (ith-harvey) [inline test/invariants/INVARIANTS.md]: which array are you referring to here? Might want to add it wrapped in back ticks.

File: test/invariants/handlers/StandardHandler.sol [hunk 2]
```diff
@@ -27,7 +27,7 @@ contract StandardHandler is Handler {
     uint256[] public proposalsExecuted;
 
     // number of proposals that recieved a vote in the given stage
-    uint256 public screeningVotesCast;
+    // uint256 public screeningVotesCast;
```
Comment (ith-harvey) [inline test/invariants/handlers/StandardHandler.sol:30]: this line still needed?

File: test/invariants/handlers/StandardHandler.sol [hunk 3]
```diff
@@ -116,6 +118,9 @@ contract StandardHandler is Handler {
             string memory description
         ) = generateProposalParams(address(_ajna), testProposalParams);
 
+        // liimit the number of proposals created in a distribution period to 200
```
Comment (ith-harvey) [inline test/invariants/handlers/StandardHandler.sol]: NIT: typo

File: test/invariants/handlers/StandardHandler.sol [hunk 4]
```diff
@@ -116,6 +118,9 @@ contract StandardHandler is Handler {
...
+        if (standardFundingProposals[distributionId].length >= 200) return;
```
Comment (ith-harvey) [inline test/invariants/handlers/StandardHandler.sol:122]: Assume this is to control the loops in the invariant tests so they don't run for too long. Suggest that this exists in a customizable env var

**Review by ith-harvey — COMMENTED** [review]
* What is the command one would run to execute the `invariant_all` scenario?
* Shouldn't running these scenarios exist in the make file? (NIT: make file could use some alignment clean up as well)
* Noticed that in the `scenarios` folder there are some files with commented lines.. do we still need those?
* `FS7` in `invariants.md` is missing colon after invariant number.

**Review by prat-gpt — APPROVED** [review]
LGTM!

File: test/invariants/handlers/StandardHandler.sol (hunk 4 above)
Comment (MikeHathaway) [reply test/invariants/handlers/StandardHandler.sol:122]: Yes there are several things like this that could be nice to parameterize. That being said, there's currently no support for env var in these invariant tests and I think this is out of scope for this PR.

File: test/invariants/handlers/StandardHandler.sol (hunk 2 above)
Comment (MikeHathaway) [reply test/invariants/handlers/StandardHandler.sol:30]: Removed

File: test/invariants/handlers/StandardHandler.sol (hunk 4 above)
Comment (MikeHathaway) [reply test/invariants/handlers/StandardHandler.sol:122]: I added support for env variables in this PR: https://github.com/ajna-finance/ecosystem-coordination/pull/108

**Comment by MikeHathaway** [comment]
> * What is the command one would run to execute the `invariant_all` scenario?
> 
>     * Shouldn't running these scenarios exist in the make file? (NIT: make file could use some alignment clean up as well)
> 
>     * Noticed that in the `scenarios` folder there are some files with commented lines.. do we still need those?
> 
>     * `FS7` in `invariants.md` is missing colon after invariant number.

https://github.com/ajna-finance/ecosystem-coordination/pull/108 fixes the first three issues

File: test/invariants/INVARIANTS.md (hunk 1 above)
Comment (MikeHathaway) [reply test/invariants/INVARIANTS.md]: Updated the comment

File: test/invariants/handlers/StandardHandler.sol (hunk 3 above)
Comment (MikeHathaway) [reply test/invariants/handlers/StandardHandler.sol]: Fixed

**Review by ith-harvey — APPROVED** [review]
LGTM

**Pull request merged**